core/templates/pages/exploration-editor-page/editor-tab/templates/modal-templates/add-hint-modal.component.ts

```diff
@@ -73,7 +73,29 @@ export class AddHintModalComponent
   }
 
   isHintLengthExceeded(tmpHint: string): boolean {
-    return Boolean(tmpHint.length > 500);
+    // The variable chars stores the remaining characters by
+    // removing the html tags from tmpHint.
+    let charRgx = /(?<=>)([^(\n)]*?)(?=<)/g;
+    let chars: string = '';
+    if (tmpHint !== null) {
+      let charArray = tmpHint.match(charRgx);
+      if (charArray !== null) {
+        chars = charArray.join('');
+      }
+    }
+    let charCount: number = chars.length;
+    // Regex for html entities like &nbsp; &lt; etc.
+    const entityRegex = /\&(.*?);/g;
+    if (entityRegex.test(chars)) {
+      let htmlEntityArray = chars.match(entityRegex);
+      if (htmlEntityArray !== null) {
+        let htmlEntityCount: number = htmlEntityArray.length;
+        let nonEntityCount: number = chars.replace(entityRegex, '').length;
+        charCount = htmlEntityCount + nonEntityCount;
+      }
+    }
+    // Note: charCount does not include the characters from Rich Text ELements.
+    return Boolean(charCount > 500);
```

Comment:
Have you considered just extracting the textContent? It might be simpler and does what we want (i.e. decodes HTML entities, only extracts the text nodes). See https://stackoverflow.com/a/34064434

Reply:
https://user-images.githubusercontent.com/99071926/215965877-e0ef8890-67e0-459b-9b4b-c698f7003f18.mp4

@kevintab95 I tried using textContent, but as you can see in the video, each time on going to next line, it is adding more newline characters than typed. But using regex, "charCount" is giving correct count.
Please confirm is this correction still needs to be made.

Reply:
@kevintab95 PTAL

Reply:
Would stripping newlines before the "length" or count operation be helpful?

Reply:
@kevintab95 Yes I thought about that, but as you can see where I have highlighted in the console, textContent is adding a a newline as well as  a blank space, though no blank space has been typed. So it will be difficult to distinguish between blank spaces that have been really typed and those that are coming due to newline.